client/src/main.jsx

```diff
@@ -176,18 +178,23 @@ const router = createBrowserRouter([
         }),
         action: async ({ request }) => {
           const formData = await request.formData();
-          const result = await addBudget({
+          const formDataObject = {
             name: formData.get("name"),
             amount: parseInt(formData.get("amount"), 10),
             startDate: formData.get("start_date"),
             endDate: formData.get("end_date"),
             categoryId: parseInt(formData.get("category"), 10),
+          };
+          const validatedData = budgetFormValidator({
+            formData: formDataObject,
           });
 
-          if (result && typeof result.message === "string") {
-            return { error: result };
+          if (Object.keys(validatedData).length > 0) {
```

Comment:
**question**: Explain me this instruction

Reply:
Question discussed in person